tests/DynamoDbCompositeModelTest.php

```diff
@@ -150,6 +150,38 @@ public function testStaticMethods()
         ])->toArray());
     }
 
+    public function testConditionContainingCompositeIndexKey()
+    {
+        $fooItem = $this->seed([
+            'id' => ['S' => 'id1'],
+            'name' => ['S' => 'Foo'],
+            'count' => ['N' => 11],
+        ]);
+
+        $barItem = $this->seed([
+            'id' => ['S' => 'id1'],
+            'name' => ['S' => 'Bar'],
+            'count' => ['N' => 9],
+        ]);
+
+        $bazItem = $this->seed([
+            'id' => ['S' => 'id1'],
+            'name' => ['S' => 'Baz'],
+            'count' => ['N' => 10],
+        ]);
+
+        $foundItems = $this->testModel
+            ->where('id', 'id1')
+            ->where('count', '>=', 10) // Test range key support comparison operator other than EQ
+            ->get();
+
+        // If id_count_index is used, $bazItem must be the first found item
```

Comment:
@baopham hash key and range key are also mentioned as **partition** key and **sort** key.
Maybe now you can understand why results are sorted by range (sort) key 😃 
Here are the reference:
http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.Partitions.html#HowItWorks.Partitions.CompositeKey
>Data Distribution: Partition Key and Sort Key
>If the table has a composite primary key (partition key and sort key), DynamoDB calculates the hash value of the partition key in the same way as described in Data Distribution: Partition Key—but it stores all of the items with the same partition key value physically close together, ordered by sort key value.

The same applied for secondary indexes.

>As to why the default sort and that it is asc.

I haven't found a reference for this. When I do normal scan or query on table, results are sorted asc by range (sort) key.